src/nvim/screen.c

```diff
@@ -2398,6 +2400,31 @@ win_line (
     filler_lines = wp->w_topfill;
   filler_todo = filler_lines;
 
+  // Cursor line highlighting for 'cursorline' in the current window.  Not
+  // when Visual mode is active, because it's not clear what is selected
+  // then.
+  if (wp->w_p_cul && lnum == wp->w_cursor.lnum
+      && !(wp == curwin && VIsual_active)) {
+    int cul_attr = win_hl_attr(wp, HLF_CUL);
+    HlAttrs *aep = syn_cterm_attr2entry(cul_attr);
+
+    // According to #7383, we make a compromise here:
+    //  * low-priority CursorLine if only background is set (foreground is not
+    //    set)
+    //  * high-priority ("same as Vim" priority) CursorLine if foreground and
+    //    background are set
+    if (aep->rgb_fg_color == -1 && aep->cterm_fg_color == 0) {
```

Comment:
Should this boolean result be stored in a local variable outside of the for loop, to avoid redundant calls to syn_cterm_attr2entry(cul_attr)?

Reply:
`syn_cterm_attr2entry` is only called once per `win_line` and it is a very simple function that return an address. It is not in a loop as least in function `win_line`.  I don't think there will be performance issue here.